File: 0011-container-with-most-water/0011-container-with-most-water.ts
function maxArea(height: number[]): number {
    let area: number = 0;
    let len: number = height.length
    let right: number = len - 1;
    let left: number = 0;
    while (left < right) {
        let calcArea = Math.min(height[left],height[right]) * Math.abs(left - right); 
        area = calcArea > area ? calcArea: area;
        if (height[left] < height[right]) {
            left += 1;
        } else {
            right -= 1;
        }
    }
    return area;
};

File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int maxWater = 0;
        int left = 0;
        int right = height.length - 1;
        
        while (left < right) {
            int length = Math.min(height[right], height[left]);
            int breadth = right - left;
            int area = length * breadth;

            maxWater = Math.max(maxWater, area);

            if (height[left] <= height[right]) {
                left++;
            } else {
                right--;
            }
        }
        return maxWater;
    }
}